MinGW warnings

On MinGW, I'll fix them later:
```
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:41: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                                         ^
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:45: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                                             ^
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:24: warning: too many arguments for format [-Wformat-extra-args]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
/home/deen/isos/ddnet/ddnet-source/src/engine/server/databases/sqlite.cpp: In member function ‘virtual void CSqliteConnection::Print()’:
/home/deen/isos/ddnet/ddnet-source/src/engine/server/databases/sqlite.cpp:233:55: warning: the address of ‘char* sqlite3_expanded_sql(sqlite3_stmt*)’ will never be NULL [-Waddress]
  233 |         if(m_pStmt != nullptr && sqlite3_expanded_sql != nullptr)
      |                                  ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
```

```
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: In function ‘time_t filetime_to_unixtime(LPFILETIME)’:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2125:18: warning: comparison of integer expressions of different signedness: ‘time_t’ {aka ‘long long int’} and ‘ULONGLONG’ {aka ‘long long unsigned int’} [-W
sign-compare]
 2125 |         return t == li.QuadPart ? t : (time_t)-1;
      |                ~~^~~~~~~~~~~~~~
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: In function ‘void set_exception_handler_log_file(const char*)’:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:4088:50: warning: cast between incompatible function types from ‘FARPROC’ {aka ‘long long int (*)()’} to ‘BOOL (*)(const char*)’ {aka ‘int (*)(const char*)’}
[-Wcast-function-type]
 4088 |                 auto pExceptionLogFilePathFunc = (BOOL APIENTRY(*)(const char *))(GetProcAddress(gs_ExceptionHandlingModule, "ExcHndlSetLogFileNameA"));
      |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: At global scope:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:226:13: warning: ‘void logger_file_no_newline(const char*, void*)’ defined but not used [-Wunused-function]
  226 | static void logger_file_no_newline(const char *line, void *user)
      |             ^~~~~~~~~~~~~~~~~~~~~~
In function ‘void str_copy(char*, const char*, int)’,
    inlined from ‘void fs_listdir(const char*, FS_LISTDIR_CALLBACK, int, void*)’ at /home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2153:11:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2616:16: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 511 bytes from a string of length 511 [-Wstringop-
truncation]
 2616 |         strncpy(dst, src, dst_size - 1);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In function ‘void str_copy(char*, const char*, int)’,
    inlined from ‘void fs_listdir_fileinfo(const char*, FS_LISTDIR_CALLBACK_FILEINFO, int, void*)’ at /home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2207:11:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2616:16: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 511 bytes from a string of length 511 [-Wstringop-
truncation]
 2616 |         strncpy(dst, src, dst_size - 1);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/base/system.cpp
+++ b/src/base/system.cpp
@@ -223,6 +223,7 @@ static void logger_file(const char *line, void *user)
 	aio_unlock(logfile);
 }
 
+#if !defined(CONF_FAMILY_WINDOWS)
 static void logger_file_no_newline(const char *line, void *user)
 {
 	ASYNCIO *logfile = (ASYNCIO *)user;
@@ -230,8 +231,7 @@ static void logger_file_no_newline(const char *line, void *user)
 	aio_write_unlocked(logfile, line, str_length(line));
 	aio_unlock(logfile);
 }
-
-#if defined(CONF_FAMILY_WINDOWS)
+#else
 static void logger_stdout_sync(const char *line, void *user)
 {
 	size_t length = str_length(line);
@@ -2122,7 +2122,7 @@ static inline time_t filetime_to_unixtime(LPFILETIME filetime)
 	li.QuadPart -= 11644473600LL; // Windows epoch is in the past
 
 	t = li.QuadPart;
-	return t == li.QuadPart ? t : (time_t)-1;
+	return t == (time_t)li.QuadPart ? t : (time_t)-1;
 }
 #endif
 
@@ -2613,8 +2613,8 @@ void str_append(char *dst, const char *src, int dst_size)
 
 void str_copy(char *dst, const char *src, int dst_size)
 {
-	strncpy(dst, src, dst_size - 1);
-	dst[dst_size - 1] = 0; /* assure null termination */
+	dst[0] = '\0';
+	strncat(dst, src, dst_size - 1);
 	str_utf8_fix_truncation(dst);
 }
 

--- a/src/engine/client/backend_sdl.cpp
+++ b/src/engine/client/backend_sdl.cpp
@@ -788,7 +788,7 @@ int CGraphicsBackend_SDL_OpenGL::Init(const char *pName, int *pScreen, int *pWid
 
 	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, g_Config.m_GfxOpenGLMajor);
 	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, g_Config.m_GfxOpenGLMinor);
-	dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
+	dbg_msg("gfx", "Created OpenGL %d.%d context.", g_Config.m_GfxOpenGLMajor, g_Config.m_GfxOpenGLMinor);
 
 	if(m_BackendType == BACKEND_TYPE_OPENGL)
 	{

--- a/src/engine/server/databases/sqlite.cpp
+++ b/src/engine/server/databases/sqlite.cpp
@@ -230,7 +230,11 @@ extern char *sqlite3_expanded_sql(sqlite3_stmt *pStmt) __attribute__((weak)); //
 
 void CSqliteConnection::Print()
 {
-	if(m_pStmt != nullptr && sqlite3_expanded_sql != nullptr)
+	if(m_pStmt != nullptr
+#if defined(__GNUC__) && !defined(__MINGW32__)
+		&& sqlite3_expanded_sql != nullptr
+#endif
+	)
 	{
 		char *pExpandedStmt = sqlite3_expanded_sql(m_pStmt);
 		dbg_msg("sql", "SQLite statement: %s", pExpandedStmt);

